source/main.lua

```diff
@@ -123,22 +126,47 @@ function drawBase()
     groundSprite:add()
 end
 
-function createObstacles()
+function createObstacles ()
     local baseX = 400 -- this is the edge of the playdate screen, but it could be something else
+    local randNum = 0
+
 
     for i = 1, groundObstacleCount, 1 do
+        -- generate a random number between 0 and 100
+        randNum = math.random(0, 100)
+
+        -- set the y value based on the random number
+        -- 30% chance of y value being at 140 (above ground) 
+        -- 70% chance of y value being at 160 (at ground level)
+        local valueY = 160
+        if randNum >= 70 then
+            valueY = 140
+        end
+
         -- generate x coords for obstacles
         groundObstacleXValues[i] = math.random(baseX, baseX + 50)
         baseX += 175
 
         -- create obstacle if not exist
         if groundObstacles[i] == nil then
-            groundObstacles[i] = gfx.sprite.new(groundObstacleImage)
+            -- Only display tall obstacle once per round
+            if not tallObstacle then
+                -- 5% chance of obstacle being taller, so use taller sprite and adjust position
+                if randNum >= 95 then
```

Comment:
this is more up for debate, but i think that tall obstacles should be more common. not excessively so, and maybe still less common than the floating ones, but 5% feels too low to me

Reply:
honestly maybe even having something like one obstacle in each round is guaranteed to be tall could be good

Reply:
Actually, grounded obstacles should not be tall. This seems to occur at random after a tall obstacle is spawned in and I am not quite sure why. I'm thinking it is for some reason substituting the tall obstacle image in place of the default one, perhaps when maintaining the table. As we are slowing development on this for now, I am going to just remove the tall obstacles and just keep them all the same size.

Reply:
fair enough

Reply:
> Actually, grounded obstacles should not be tall. This seems to occur at random after a tall obstacle is spawned in and I am not quite sure why.

The code for determining a tall obstacle is all in the if statement that checks if groundObstacles[i] == nil (line 151). So once one round has passed this code will never run again (or at least not until a reset). Hence, obstacles chosen as tall on the first run will always be tall and lower (due to the y value being reset at the beginning of the loop). The simplest solution would be to remove that if check, I'm not entirely sure if I like that but it might actually be appropriate here since any obstacle can either be tall or short, so it'd be hard to optimize that.

At the end of the loop tallObstacle should be set back to false, otherwise you won't get another tall obstacle in a future round until a reset. Also in obstacleLogic() there's now a bug due to recreating each obstacle where the last obstacle does not get removed. This can be fixed by adding obstacleSprite:remove() to the if statement at line 380 (I'd also change the position in the condition from 0 to -7, otherwise the obstacle will just abruptly disappear).

If you still don't want to have tall obstacles that's fine, I just don't know if you want to give it another try with this information (note that I haven't looked into this too deeply, I'm not sure if what I've said is all that's needed or the best solution). We have paused features but since you already have the code for this you might as well finish it up. Reply here what you want to do, I will look at the rest of the PR afterwards.

Side note, in the future if you run into an issue like this you should get into the voice chat to look at it with other people. Often times it's better to have at least one other person to discuss with rather than doing it independently. I even think that doing a quick voice chat before opening a PR would be beneficial (since it would get another set of eyes on the code and you can usually find mistakes just by talking out loud).